integration-cli/docker_cli_v2_only_test.go

```diff
@@ -65,6 +65,49 @@ func (s *DockerRegistrySuite) TestV2Only(c *check.C) {
 	s.d.Cmd("pull", repoName)
 }
 
+// TestUserAgentPassThroughOnPull verifies that when an image is pulled from
+// a registry, the registry should see a User-Agent string of the form
+// "[client UA] upstream-ua/end; [docker engine UA]"
+func (s *DockerRegistrySuite) TestUserAgentPassThroughOnPull(c *check.C) {
```

Comment:
Please create a new file for this test, since it's not related to the `--disable-legacy-registry` option.